kroxylicious/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -37,17 +37,19 @@ public class VirtualCluster implements ClusterEndpointConfigProvider {
     private final boolean logNetwork;
 
     private final boolean logFrames;
-    private final ClusterEndpointConfigProvider endpointProvider;
+
+    // There appear to be a Jackson defect that is triggered if this field name matches the name of the ctor's parameter causing it to ignore the converter.
+    private final ClusterNetworkAddressConfigProvider provider;
```

Comment:
I discovered what appears to be a jackson issue here where the convertor gets ignored. It must be taking the type information from the wrong place.  I intend to create a reproducer and raise issue/fix if I can.